app/src/Pages/DetailsPage/Details.tsx

```diff
@@ -38,14 +63,44 @@ function Details() {
     console.log("price history state after adding stuff in: ", priceHistory); 
   }
 
+  /*
+  Function to be called when new card rating is posted. 
+  Appends new card rating entry to the card rating array. 
+  Then, React will automatically re-render the CardDescription component
+  to include the newly computed averages of all the card ratings.
+  */
+  const handleNewCardRatingSubmission = (newCardRatingSubmission: CardRating) => {
+    console.log("card rating state before adding stuff in: ", cardRating); 
+    setCardRating(cardRating => [newCardRatingSubmission, ...cardRating]); 
```

Comment:
This is super clean, I've never used this syntax so I will have to look into how it works.